src/test/test-strv.c

```diff
@@ -659,6 +659,22 @@ static void test_strv_foreach_pair(void) {
                 assert_se(streq(*x, *y));
 }
 
+static void test_strv_foreach_index(void) {
+        _cleanup_strv_free_ char **a;
+        unsigned i = 0, j;
+        char **check;
+
+        log_info("/* %s */", __func__);
+
+        a = strv_new("one", "two", "three");
+        assert_se(a);
+
+        STRV_FOREACH_INDEX(check, a, j) {
+                assert_se(i == j);
+                assert_se(streq(*check, input_table_multiple[i++]));
+        }
```

Comment:
Hmm, I'd prefer to write such iteration as:
```C
for (size_t i = 0; s && s[i]; i++)
   ... s[i]
```
This avoids the need to define two variables and the use of `char**`. The compiler will optimize this just as well or even better.

Reply:
Ok, changed and dropped new helper